Automattic/pocket-casts-ios
# Refactor the use of the AutoDownloadStatus enums

| 📘 Part of: # |  
|:---:|

Fixes # 


On this PR we use the enum directly on the model type and avoid the uses of raw value and casting in the codebase. I also found that FMDatabase can support the use of Enum directly on their statements if they are a base type. Ex: Int32, Int, etc..

One of the things that was blocking it was that our base episode protocol was marked as ObjC. I thought there was some reason for the use of `@objc` tag on the protocols, but after removal I didn't get any kind of error.

If someone knows the reason why those protocols are marked @objc please comment bellow.

## To test

- Start the app
- Download some episode
- Check if download status updates are done correctly and reported in the app correctly.
- Check if no crashes or warnings show up in the console.

## Checklist
